### DISCUSSION TITLE
Using ESP32 ethernet (WT32-ETH01) with STM32 and OTA

### DISCUSSION THREAD

**Discussion by Gifford47** [arendst/Tasmota#21789] [Support request]
Hi!

I want to connect an ESP32 (WT32-ETH01) with a STM32 (black pill). 
I currently have an ETH01 board connected to the STM32 (TX, RX, NRST, Boot0) and [configured ](https://templates.blakadder.com/wireless_tag_WT32-ETH01.html) the ETH01 board as follows:
```
{"NAME":"WT32_ETH01","GPIO":[1,1,1,1,1,1,0,0,1,0,1,1,3840,576,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,1,1,0,1,1,0,0,1],"FLAG":0,"BASE":1}
``` 
I then wanted to configure SHD BOOT0 and SHD Reset [like a Shelly Dimmer](https://templates.blakadder.com/shelly_Dimmer_2.html) in the module configuration but then I saw, that gpio 5568,5600 on ESP32 are used for ETH MDC and ETH MDIO instead of SHD BOOT0 and SHD Reset (on normal Tasmota build).
I´ve build a custom firmware to use stm-flasher1.0 for STM OTA feature but this is useless if the pins are not configured for this.

I there a simple way to use an ESP32 with ethernet and a STM32 including OTA feature? Does anyone know what the code change for this could look like?

**Comment by sfromis** [comment]
Don't worry about what pins you use, or which are already in use. You can just config any two usable gpios, as ESP32 does **not** have locked pins for most functions, only Analog and Ethernet ar tied to certain pins.

**Reply by Gifford47** [reply]
Ok, but I haven't a selection/option in the drop-down field to choose SHD Boot0 or reset. I think it's not coded for esp32...

**Reply by sfromis** [reply]
Well, as the Shelly Dimmer is not sold with ESP32, Tasmota does not include this specialized support with ESP32 devices. And if you modify Tasmota to include it (look for where `USE_SHELLY_DIMMER` is used), it is untested.

Also, the details seem to depend on what STM32 variant you have, meaning that you may have an additional thing there to set up for your variant.

**Reply by Gifford47** [reply]
I´ve compiled latest tasmota32 environment with ```USE_SHELLY_DIMMER``` and got the following error: ```tasmota/include/tasmota_compat.h:23:10: fatal error: esp8266toEsp32.h: No such file or directory```

How can I include the missing lib?
